classes/abstracts/ActionScheduler_Store.php

```diff
@@ -208,6 +209,40 @@ abstract public function get_claim_id( $action_id );
 	 */
 	abstract public function find_actions_by_claim_id( $claim_id );
 
+	/**
+	 * Check if group name is prefixed for exclussion.
+	 *
+	 * @param string $slug Group slug name.
+	 * @return boolean
+	 */
+	protected static function group_has_ignore_prefix( $slug ) {
+		return self::IGNORE_GROUP_FLAG === substr( $slug, 0, strlen( self::IGNORE_GROUP_FLAG ) );
+	}
+
+	/**
+	 * Mark group for exclussion by prefixing with '-'.
+	 *
+	 * @param string $slug Group slug name.
+	 * @return string
+	 */
+	public static function mark_group_for_exclussion( $slug ) {
+		return $slug ? self::IGNORE_GROUP_FLAG . $slug : $slug;
+	}
+
+	/**
+	 * Sanitize group name.
+	 *
+	 * @param string $slug Group slug name.
+	 * @return string
+	 */
+	public static function sanitize_group_name( $slug ) {
+		// removing the ignored group prefix from slug if present.
+		$slug = preg_replace( '/(' . preg_quote( self::IGNORE_GROUP_FLAG, null ) . ')?(.*)/m', '$2', $slug );
+
+		// further sanitisation here with `sanitize_title` maybe?
+		return $slug;
```

Comment:
Perhaps we should also anchor `^`the regex? Re sanitization; I wonder if `esc_sql` would (also) make sense?

Reply:
Added the ^ regex change as well as the `esc_sql` filtering